### PR TITLE
Invoice unpaid NFT migration

### PR DESCRIPTION
**Pull request by vimukthi-git**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: <Github link>

### CODE REVIEW

File: httpapi/userapi/service.go [hunk 1]
```diff
@@ -2,6 +2,14 @@ package userapi
...
 import (
 	"context"
+	"fmt"
+
+	"github.com/centrifuge/go-centrifuge/config"
+	"github.com/centrifuge/go-centrifuge/contextutil"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/nft"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
```
Comment (charleenfei) [inline httpapi/userapi/service.go]: import spacing

File: httpapi/userapi/service_test.go [hunk 2]
```diff
@@ -4,9 +4,14 @@ package userapi
...
 import (
 	"context"
+	"fmt"
 	"testing"
 	"time"
 
+	"github.com/centrifuge/go-centrifuge/config/configstore"
+	"github.com/centrifuge/go-centrifuge/contextutil"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
```
Comment (charleenfei) [inline httpapi/userapi/service_test.go]: import spacing

File: httpapi/userapi/invoice_api_test.go [hunk 3]
```diff
@@ -11,6 +11,14 @@ import (
 	"net/http/httptest"
 	"testing"
 
+	"github.com/centrifuge/go-centrifuge/nft"
+
+	"github.com/centrifuge/go-centrifuge/config/configstore"
```
Comment (charleenfei) [inline httpapi/userapi/invoice_api_test.go]: import spacing

File: httpapi/userapi/service.go [hunk 4]
```diff
@@ -243,3 +251,56 @@ func (s Service) RevokeRelationship(ctx context.Context, docID []byte, req Share
...
+		GrantNFTReadAccess:       true,
+		SubmitNFTReadAccessProof: true,
+		SubmitTokenProof:         true,
+	}
+
+	return s.coreAPISrv.MintNFT(ctx, nreq)
+}
+
+// GetRequiredInvoiceUnpaidProofFields returns required proof fields for an unpaid invoice mint
+func GetRequiredInvoiceUnpaidProofFields(ctx context.Context) ([]string, error) {
```
Comment (vedhavyas) [inline httpapi/userapi/service.go]: do we need this to be exported ?

File: httpapi/userapi/types.go [hunk 5]
```diff
@@ -46,6 +46,24 @@ type TransferDetailListResponse struct {
 	Data   []transferdetails.Data `json:"data"`
 }
 
+// NFTMintInvoiceUnpaidRequest is the request for minting an NFT for an unpaid NFT
+type NFTMintInvoiceUnpaidRequest struct {
+	// Invoice Document identifier
+	DocumentId string `json:"document_id"`
```
Comment (vedhavyas) [inline httpapi/userapi/types.go]: DocumentID

File: httpapi/userapi/types.go [hunk 6]
```diff
@@ -46,6 +46,24 @@ type TransferDetailListResponse struct {
...
+	// Deposit address for NFT Token created
+	DepositAddress string `json:"deposit_address"`
+}
+
+// ResponseHeader header with job id
+type ResponseHeader struct {
+	JobId string `json:"job_id"`
```
Comment (vedhavyas) [inline httpapi/userapi/types.go]: JobID

File: httpapi/coreapi/bootstrapper.go [hunk 7]
```diff
@@ -27,7 +27,7 @@ func (b Bootstrapper) Bootstrap(ctx map[string]interface{}) error {
 		return errors.New("failed to get %s", jobs.BootstrappedService)
 	}
 
-	nftSrv, ok := ctx[bootstrap.BootstrappedInvoiceUnpaid].(nft.InvoiceUnpaid)
```
Comment (vedhavyas) [inline httpapi/coreapi/bootstrapper.go]: 👍

File: httpapi/userapi/types.go [hunk 8]
```diff
@@ -46,6 +46,24 @@ type TransferDetailListResponse struct {
...
+}
+
+// ResponseHeader header with job id
+type ResponseHeader struct {
+	JobID string `json:"job_id"`
+}
+
+// NFTMintResponse is response from user api NFT minting
+type NFTMintResponse struct {
+	Header *ResponseHeader `json:"header"`
```
Comment (vedhavyas) [inline httpapi/userapi/types.go]: Sorry, missed in the last read through. I think value is performant here instead of a pointer. Not a blocker though.

**Comment by vimukthi-git** [comment]
@vedhavyas sorry, I missed your second set of comments :|

**Pull request merged**